### PR TITLE
Replaced if statements with "switch"-like dict to reduce complexity

### LINKED ISSUE #28: "render_question" method in callisto/delivery/report_delivery.py is too complex
Cyclomatic complexity is too high in method render_question. (17)

https://codeclimate.com/github/SexualHealthInnovations/callisto-core/callisto/delivery/report_delivery.py#issue_5769b1b54a3c920001de113f

### PR DESCRIPTION
**Pull request by keisetsu**
fixes #28

### CODE REVIEW

File: callisto/delivery/report_delivery.py [hunk 1]
```diff
@@ -244,11 +247,24 @@ def render_question(self, question):
                     for q in form:
                         self.render_question(q)
                     self.pdf_elements.append(Indenter(left=-12))
-            else:
-                self.add_question(prompt)
-                self.pdf_elements.append(Indenter(left=12))
-                self.add_question('<i>None added</i>')
-                self.pdf_elements.append(Indenter(left=-12))
+
+        def _render_default():
```
Comment (kelseyq) [inline callisto/delivery/report_delivery.py]: this logic isn't quite right--what you've put in render_default should be part of render_formset, it's what gets rendered when there are no forms in the formset

File: callisto/delivery/report_delivery.py [hunk 2]
```diff
@@ -244,11 +247,24 @@ def render_question(self, question):
...
+            prompt = question.get('prompt').capitalize()
+            self.add_question(prompt)
+            self.pdf_elements.append(Indenter(left=12))
+            self.add_question('<i>None added</i>')
+            self.pdf_elements.append(Indenter(left=-12))
+
+        render_functions = {
```
Comment (kelseyq) [inline callisto/delivery/report_delivery.py]: this is missing MultipleChoice and Date question types

**Comment by kelseyq** [comment]
The idea here looks solid to me, but I'm wary of merging it because it has to be tested manually right now. Any interest in tackling https://github.com/SexualHealthInnovations/callisto-core/issues/19 to go along with it?

**Comment by keisetsu** [comment]
Wow, yeah those were pretty bad. Starting with some tests would have been a good idea! At least I'll be more careful with the tests.

I'll go ahead and fix these problems now, but I'll get working on #19 when I get a chance.

File: callisto/delivery/report_delivery.py (hunk 2 above)
Comment (keisetsu) [inline callisto/delivery/report_delivery.py]: Just to make sure I didn't miss something further: I did miss Date here, but I do have RadioButton in the render_function dict. I put the keys in alpha order, so it's not next to Checkbox, but it still will call _render_multiple_choice. Right?

**Comment by keisetsu** [comment]
Since there is no "default" question type, should I raise an error or just quietly ignore the possibility of another type of field?

**Comment by kelseyq** [comment]
Hey Jesse--

I'm so sorry for the delay on this! I'm getting caught back up now and I wanted to let you know I'm still definitely interested in finishing this fix now that #119 tests this functionality.

--Kelsey

**Comment by lynnco** [comment]
Heya @keisetsu ! Can you give me edit permissions on your branch so I can update this PR? https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

Or you can handle that yourself, whichever ^_^

**Comment by keisetsu** [comment]
There you go! I hope you don't mind if I leave it to you: it's been such a long time since I've taken a look at your code!

**Comment by lynnco** [comment]
Closing this since its not worth resolving the merge errors introduced by https://github.com/SexualHealthInnovations/callisto-core/pull/202

I'll circle back around and address the issue eventually though

**Pull request closed without merging**